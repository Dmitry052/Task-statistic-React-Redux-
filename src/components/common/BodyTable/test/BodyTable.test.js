import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { BodyTable } from './../index';
import {
  formatTime,
  countryFormat
} from './../../../main/owner/ContentTable/utils';
import pres from './../../../main/owner/ContentTable/localFormat';

configure({ adapter: new Adapter() });

describe('<BodyTable />', () => {
  const bodyTable = [
    { bill_date: 'date', text: '-', precision: null, nowrap: null },
    { coins: 'string', text: '-', precision: null, nowrap: null },
    { tokens_sold: 'number', text: '-', precision: null, nowrap: null },
    { country: 'country', text: '-', precision: null, nowrap: null },
    { some_field: 'not_key', text: '-', precision: null, nowrap: null },
    { some_field2: 'number', text: '-', precision: 1, nowrap: null },
    { some_field3: 'not_key', text: 'not key in itemTR', precision: 1, nowrap: null }
  ];
  const itemTR = {
    bill_id: 1,
    bill_date: '2018-04-16T11:07:27.000Z',
    tokens_sold: '971',
    coins: 'COUPON',
    email: 'billhegarty@postmaster.co.uk',
    kyc_statuses: 'NEW',
    country: 'ru',
    some_field: 'not switch case',
    some_field2: '23233,4444'
  };
  const index = 1;
  const tree = shallow(
    <BodyTable
      bodyTable={bodyTable}
      itemTR={itemTR}
      index={index}
      formatTime={formatTime}
      pres={pres}
      countryFormat={countryFormat}
    />
  );

  it('should contain 5 tr elements', () => {
    expect(tree.find('tr').children().length).toBe(6);
  });

  it('Matches the snapshot BodyTable component', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });
});
