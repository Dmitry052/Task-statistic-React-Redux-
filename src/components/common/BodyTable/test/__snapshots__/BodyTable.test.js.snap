// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyTable /> Matches the snapshot BodyTable component 1`] = `
<React.Fragment>
  <tr
    key="tr-itemTR-1"
  >
    <td
      key="td-itemTD-0"
      nowrap="false"
    >
      2018-04-16
    </td>
    <td
      key="td-itemTD-1"
      nowrap="false"
    >
      COUPON
    </td>
    <td
      key="td-itemTD-2"
      nowrap="false"
    >
      971
    </td>
    <td
      key="td-itemTD-3"
      nowrap="false"
    >
      <div>
        <ReactCountryFlag
          code="ru"
          svg={true}
        />
          
        ru
      </div>
    </td>
    <td
      key="td-itemTD-5"
      nowrap="false"
    >
      NaN
    </td>
    <td
      key="td-itemTR-6"
      nowrap="false"
    />
  </tr>
</React.Fragment>
`;
